Getinvolved (#83)
pharmaR/pharmaR.github.io: content/contribute.md

# Contributing to the R Validation Hub

## Mailing List

> For occasional news updates, you can join our [mailing list](https://lists.r-consortium.org/g/RConsortium-Validation-Hub). This will enable you and your colleagues to receive notifications of key updates/software releases, blog posts and **invites to all-hands Hub meetings**.  

Reviewer: This may be outside the scope of this PR, but I just clicked on this [mailing list](https://lists.r-consortium.org/g/RConsortium-Validation-Hub) link and found the site to be pretty out of date & difficult to navigate. The executive committee member list is from 2019, and I found that most of the page has nothing to do with a mailing list at all. That is, I found it difficult to find where I could actually sign up for the mailing list. After a few minutes, I found a little button at the bottom of the page labeled `+ Join this group` but I'd argue I exerted way too much effort to achieve my goal (of signing up). A simple solution: create a new header titled "Join our mailing list" just before the sentence that reads: 

> "To manage your subscriptions to the public R Consortium mailing lists, please visit https://lists.r-consortium.org/g/Rconsortium-main/subgroups."

Author: Good catch!  There's a limit to what you can do but I have tidied it a bit, removing a lot of the repeated detail from the website to reduce maintenance burden.  Realistically, people will generally only find the mailing list page by visiting the website.

## Active Participation

> If you are willing and able to commit more time to the R Validation Hub then there are several options depending on your skills and interests.  If you do not feel you have the necessary technical skills to contribute then please don't let that stop you getting in touch and sharing ideas to support the implementation of a risk-based R package assessment.

Reviewer: Consider inserting the word "mission" in to this sentence so it reads: "...commit more time to the R Validation Hub's **mission**, then there are several options..."

Consider adding a comma after the word "contribute"

> If you are willing and able to commit more time to the R Validation Hub then there are several options depending on your skills and interests.  If you do not feel you have the necessary technical skills to contribute then please don't let that stop you getting in touch and sharing ideas to support the implementation of a risk-based R package assessment.

Author: ```suggestion
If you are willing and able to commit more time to the R Validation Hub's mission, then there are several options depending on your skills and interests.  If you do not feel you have the necessary technical skills to contribute then please don't let that stop you getting in touch and sharing ideas to support the implementation of a risk-based R package assessment.
```

In each case you can get involed either by contacting the stream leads via the relevant GitHub page, or by sending a message to us via the contact form on our [Contact Us](/contact/) page.

### The {riskmetric} R Package

> The {riskmetric} package collects metrics that support the idea of a risk-based evaluation of R packages.  The package collects and summarises various metrics and provides the means to quantify the risk of a package via a weighted risk score.  It also serves as the backbone to the Risk Assessment shiny application.  For more information see the [package site](https://pharmar.github.io/riskmetric/) or [GitHub page](https://github.com/pharmaR/riskmetric).

Reviewer: ```suggestion
The {riskmetric} package collects curated metrics that support the idea of a risk-based evaluation of R packages. The package not only collects and summarises these metrics, but also provides a means to quantify the risk via a weighted risk score.  {riskmetric} also serves as the backbone to the Risk Assessment shiny application.  For more information see the [package site](https://pharmar.github.io/riskmetric/) or [GitHub page](https://github.com/pharmaR/riskmetric).
```

**Current Focus**

> The {riskmetric} team are currently looking to build features to support the idea of package cohorts.  We are also looking to refine some of the meta-information collected by riskmetric to better-serve the risk assessment application.  The team also have a [backlog of other open issues](https://github.com/pharmaR/riskmetric/issues), including additional metrics that they would like to explore.

Reviewer: ```suggestion
The {riskmetric} team are looking to build features to support the idea of package cohorts. Another priority is to refine some meta-information collected by {riskmetric} to better-serve the risk assessment application.  The development team also has a [backlog of other open issues](https://github.com/pharmaR/riskmetric/issues), including additional metrics that they would like to explore.
```

**Key Skills Required**

> A fundamental understanding of R packages is essential for anyone looking to support the {riskmetric} team.  Package development experience is also highly beneficial.  

Reviewer: ```suggestion
A fundamental understanding of R packages is essential for anyone looking to support the {riskmetric} team. Package development experience is also highly beneficial.  
```

### The Risk Assessment Application

> The risk assessment shiny app is an extension of the {riskmetric} R package and provides a graphical interface to the {riskmetric} functionality.  It provides further exploratory capabilities in addition to the numeric metrics and improves ease-of-use for non-technical users. For more information see the [package GitHub page](https://github.com/pharmaR/risk_assessment).

Reviewer: Well said!

> The risk assessment shiny app is an extension of the {riskmetric} R package and provides a graphical interface to the {riskmetric} functionality.  It provides further exploratory capabilities in addition to the numeric metrics and improves ease-of-use for non-technical users. For more information see the [package GitHub page](https://github.com/pharmaR/risk_assessment).

Reviewer: ```suggestion
The risk assessment shiny app is an extension of the {riskmetric} R package, providing a graphical interface to the {riskmetric} functionality and a means to collaborate when assessing packages risk within an organization. It provides further exploratory capabilities in addition to the numeric metrics and improves ease-of-use for non-technical users. For more information see the [package GitHub page](https://github.com/pharmaR/risk_assessment).
```

**Current Focus**

In addition to working with the {riskmetric} team to better-support cohorts of packages and multiple package versions, the team are building up metric and package exploration features.  This includes links to source information such as vignettes, and the ability to explore package documentation and tests.

**Key Skills Required**

> Shiny app development experience is highly desirable.

Reviewer: Consider amending to: "Shiny app development experience is highly desirable; specifically, experience with shiny modules and the {golem} package would be beneficial, but is not required."

### Regulatory R Package Repository

This is a brand new (Q3 2022) initiative that is just getting off the ground.  The aim is to create a cross-pharma regulatory package repository.  For the latest, please check out the [GitHub page](https://github.com/pharmaR/regulatory-r-repo-wg).

**Current Focus**

> The first milestone (projected three months) will be primarily focused around building an initial core team and more thoroughly scoping the project.

Reviewer: Three months from when? Maybe it'd be better to say: "(projected by Dec 2022)"?

**Key Skills Required**

To get this project off the ground requires a variety of skills.  The aim is to assemble a team with backgrounds spanning:

- R package development (specifically testing & builds)
- Infrastructure & administration (containers, CI/CD, repository hosting)
- Front end & API serving

## How else can I help?

> If you have tried to implement any of the ideas the the R Validation Hub have outlined then we would love you to submit a case study to our [Case Study](https://github.com/pharmaR/case_studies) catalogue on GitHub.  USe our mailing list or GitHub issues to let us know of your intentions and we'll work with you to upload your written case study.  We're also pleased to share case studies verbally with the wider community.  You can find videos of previous presentations on our [case studies](/casestudies/) page.

Reviewer: Typo in first sentence: "the the"

Typo: capital S in: "USe our mailing list"

Without clicking on the case studies link, are you asking if people have used the RvalHub's **ideas** or one of their **tools**, like `{riskmetric}` and risk assessment? Or are you asking what methods their organizations use for assessing risk if **NOT** leveraging these RvalHub tools? It's unclear so I'm not how I can help... if I were a reader.

> If you have tried to implement any of the R Validation Hub's ideas, or used any of our tools, or even if you've decided to go own way, then we would love you to submit a case study describing what you have done to implement a GxP-compliant R environment to our [Case Study](https://github.com/pharmaR/case_studies) catalogue on GitHub.  Use our mailing list or GitHub issues to let us know of your intentions and we'll work with you to upload your written case study.  We're also pleased to share case studies verbally with the wider community.  You can find videos of previous presentations on our [case studies](/casestudies/) page.

Reviewer: ```suggestion
If you have tried to implement any of the R Validation Hub's ideas, used our tools, or even if you've designed your own, we would love for you to submit a case study describing your actions to implement a GxP-compliant R environment to our [Case Study](https://github.com/pharmaR/case_studies) catalogue on GitHub.  Use our mailing list or GitHub issues to let us know of your intentions and we'll work with you to upload your written case study.  We're also pleased to share case studies verbally with the wider community.  You can find videos of previous presentations on our [case studies](/casestudies/) page.
```
